utils/estimated_histogram.hh

```diff
@@ -73,40 +73,59 @@ struct estimated_histogram {
 
     int64_t _count = 0;
     int64_t _sample_sum = 0;
+    int64_t _min = 0;
+    int64_t _max = 0;
 
     estimated_histogram(int bucket_count = 90) {
 
         new_offsets(bucket_count);
         buckets.resize(bucket_offsets.size() + 1, 0);
     }
 
-    seastar::metrics::histogram get_histogram(size_t lower_bucket = 1, size_t max_buckets = 16) const {
+    /*
+     * \brief get an accumulative histogram
+     * When getting an accumulative histogram, you can set what is the first bucket to report
+     * what is the number of buckets
+     * and what is the ratio between bucket limits.
+     * Setting the bucket_ratio for 1 or less, will use the buckets as-is
+     */
+    seastar::metrics::histogram get_histogram(size_t lower_bucket = 1, size_t max_buckets = 20, double bucket_ratio = 1.44) const {
         seastar::metrics::histogram res;
         res.buckets.resize(max_buckets);
-        int64_t last_bound = lower_bucket;
+        double last_bound = lower_bucket;
+        double current_bound = 0;
         uint64_t cummulative_count = 0;
         size_t pos = 0;
 
         res.sample_count = _count;
         res.sample_sum = _sample_sum;
         for (size_t i = 0; i < res.buckets.size(); i++) {
             auto& v = res.buckets[i];
-            v.upper_bound = last_bound;
 
             while (bucket_offsets[pos] <= last_bound) {
                 cummulative_count += buckets[pos];
+                current_bound = bucket_offsets[pos];
                 pos++;
             }
-
+            if (bucket_ratio <= 1) {
+                last_bound = bucket_offsets[pos];
+            } else {
+                last_bound *= bucket_ratio;
+            }
```

Comment:
I don't understand this bucket_ratio < 1 thing. Isn't there a bucket_ratio that corresponds to the current code? If so, it can be the default.

Reply:
It eliminates all the ratio calculation, so it allows the caller to say: give me the maximal precision available.
If we know that we want the maximal precision, why do the calculation? instead, just take the value.

Reply:
I don't understand what's the maximal precision, since we create buckets in any case.

Reply:
We start with the base estimated histogram (typically, uses a ratio of 1.2 and 92 bucket but can use anything else).
When we report it, we use an accumulative histogram with an (optional) different ratio (i.e. different precision).
So the user of the accumulative histogram (Prometheus) can choose to limit the size of the histogram it will get, or ask for the highest precision histogram available.
The idea is to allow requesting the histogram without reducing the number of buckets.

A valid alternative is to report it by default.
If we are reporting all the buckets, there is no point in calculating the buckets limits, instead, we can just copy the values.

Reply:
I still don't understand what bucket_ratio <= 1 means. Is it a way to say "change the method of calculating ratios to something else"? If so it is a bad method because no one will be able to guess it.

Reply:
I see there's a comment that says so. But it's a bad API. Better is to use a function with a different name (internally they can share implementations).

Reply:
I'll replace it all anyhow, this is a step that keeps the current API.